runtime/parachains/src/session_info.rs

```diff
@@ -196,10 +210,20 @@ impl<T: Config> Pallet<T> {
 			dispute_period,
 		};
 		Sessions::<T>::insert(&new_session_index, &new_session_info);
-		SessionExecutorParams::<T>::insert(
-			&new_session_index,
-			ExecutorParams::from(&EXECUTOR_PARAMS[..]),
-		);
+
+		let new_executor_params = if let Some(executor_params) = <PendingExecutorParams<T>>::get() {
+			<PendingExecutorParams<T>>::put(&None);
+			executor_params
+		} else {
+			let current_session_index = <shared::Pallet<T>>::session_index();
+			if let Some(executor_params) = <SessionExecutorParams<T>>::get(current_session_index) {
+				executor_params
+			} else {
+				unreachable!("Current session executor params must always be present");
```

Comment:
There is no advantage here of having a potential panic (which must be avoided in consensus code at all costs, see [SE](https://substrate.stackexchange.com/questions/317/when-to-use-expect)).  

Rather use `frame_support::defensive!("Current session executor params must always be present")` and skip the insertion below, which should incur no harm - besides the pending params upgrade being discarded.